vortex-array/src/arrays/primitive/compute/between.rs

```diff
@@ -65,7 +90,7 @@ where
             let i = unsafe { *slice.get_unchecked(idx) };
             lower_fn(lower, i) & upper_fn(i, upper)
         }),
-        arr.validity().clone(),
+        arr.validity().clone().merge_nullability(nullability),
```

Comment:
if nullability == Nullability::NonNullable { arr.validity().clone() } else { arr.validity().clone().into_nullable() }

Reply:
So you are saying you don't want to add a new method?